libs/orchestrator-ui-components/src/lib/hooks/useDataDisplayParams.ts

```diff
@@ -0,0 +1,64 @@
+import {
+    NumberParam,
+    ObjectParam,
+    useQueryParams,
+    withDefault,
+} from 'use-query-params';
+
+import { GraphQLSort, GraphqlFilter } from '../types';
+import { DEFAULT_PAGE_SIZE } from '../components';
+
+export type DataDisplayParams<Type> = {
+    pageSize: number;
+    pageIndex: number;
+    sortBy?: GraphQLSort<Type>;
+    filterBy?: GraphqlFilter<Type>[];
+};
+
+export interface DataDisplayReturnValues<Type> {
+    dataDisplayParams: DataDisplayParams<Type>;
+    setDataDisplayParam: <PropKey extends keyof DataDisplayParams<Type>>(
+        prop: PropKey,
+        value: DataDisplayParams<Type>[PropKey],
+    ) => void;
+}
+
+export const useDataDisplayParams = <Type>(
+    initialParams: Partial<DataDisplayParams<Type>>,
+): DataDisplayReturnValues<Type> => {
+    const defaultParams = {
+        pageSize: DEFAULT_PAGE_SIZE,
+        pageIndex: 0,
+        ...initialParams,
+    };
+
+    const [dataDisplayParams, setDataDisplayParams] = useQueryParams({
```

Comment:
Just checking, is it not possible to specify the type "DataDisplayParams<Type>" directly after the useQueryParams?
`useQueryParams<DataDisplayParams<Type>>` to avoid the type cast in the return statement.

Reply:
I have tried to pass a type hint to useQueryParams so it would return the right thing right away but couldn't get that to work. Baring that I think the most elegant way is to typecast at the end since you see what going on a bit easier